pkg/api/util/db_test.go

```diff
@@ -682,3 +682,160 @@ func TestDeleteFileShareAclDBEntry(t *testing.T) {
 		}
 	})
 }
+
+func TestCreateFileShareAclDBEntry(t *testing.T) {
+	var in = &model.FileShareAclSpec{
+		BaseModel: &model.BaseModel{
+			Id: "6ad25d59-a160-45b2-8920-211be282e2df",
+		},
+		Description: "This is a sample Acl for testing",
+		ProfileId:   "1106b972-66ef-11e7-b172-db03f3689c9c",
+		Type: "ip",
+		AccessCapability: []string{"Read", "Write"},
+		AccessTo: "10.32.109.15",
+		FileShareId: "d2975ebe-d82c-430f-b28e-f373746a71ca",
+	}
+
+	t.Run("Everything should work well", func(t *testing.T) {
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		var expected = &SampleFileSharesAcl[2]
+		result, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		if err != nil {
+			t.Errorf("failed to create fileshare err is %v\n", err)
+		}
+		assertTestResult(t, result, expected)
+	})
+
+	t.Run("If profile id is empty", func(t *testing.T) {
+		in.ProfileId = ""
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := "profile id can not be empty when creating fileshare acl in db!"
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid Access Type", func(t *testing.T) {
+		in.ProfileId, in.Type = "d2975ebe-d82c-430f-b28e-f373746a71ca", "system"
```

Comment:
why does profile id need to be set here ?

Reply:
Because of local scope of variable, it has been set to empty in the previous test case, now it should be set with proper value to execute next condition